pkg/controller/config-controller.go

```diff
@@ -267,27 +268,65 @@ func (c *ConfigController) syncHandler(key string) error {
 			}
 		}
 	}
+
 	if (config.Status.UploadProxyURL != nil && url == *config.Status.UploadProxyURL) || (config.Status.UploadProxyURL == nil && url == "") {
-		//status already stores the URL
-		return nil
+		updateConfig = false
+	} else {
+		updateConfig = true
 	}
-
 	newConfig := config.DeepCopy()
-	// mutate newConfig
-	if url == "" {
-		newConfig.Status.UploadProxyURL = nil
+	if updateConfig {
+		// mutate newConfig
+		if url == "" {
+			newConfig.Status.UploadProxyURL = nil
+		} else {
+			newConfig.Status.UploadProxyURL = &url
+		}
+	}
+
+	storageClass, err := c.scratchSpaceStorageClassStatus(config)
+
+	if storageClass == config.Status.ScratchSpaceStorageClass {
```

Comment:
comparing pointer values here not string values so will never be equal.

Recommend `config.Status.ScratchSpaceStorageClass` be `string` and `scratchSpaceStorageClassStatus` returns `(string, error)`

Reply:
Done